Source/WebCore/css/calc/CSSCalcOperationNode.cpp

```diff
@@ -1273,229 +1272,11 @@ bool CSSCalcOperationNode::equals(const CSSCalcExpressionNode& exp) const
-        if (children[0] > 0)
-            return 1;
-        if (children[0] < 0)
-            return -1;
-        return children[0];
-    }
-    case CalcOperator::Mod: {
-        if (children.size() != 2)
-            return std::numeric_limits<double>::quiet_NaN();
-        float left = children[0];
-        float right = children[1];
-        if (!right)
-            return std::numeric_limits<double>::quiet_NaN();
-        if ((left < 0) == (right < 0))
-            return std::fmod(left, right);
-        return std::remainder(left, right);
-    }
-    case CalcOperator::Rem: {
-        if (children.size() != 2)
-            return std::numeric_limits<double>::quiet_NaN();
-        float left = children[0];
-        float right = children[1];
-        if (!right)
-            return std::numeric_limits<double>::quiet_NaN();
-        return std::fmod(left, right);
-    }
-    case CalcOperator::Round:
-        return std::numeric_limits<double>::quiet_NaN();
-    case CalcOperator::Up: {
-        if (children.size() != 2)
-            return std::numeric_limits<double>::quiet_NaN();
-        if (!isinf(children[0]) && std::isinf(children[1])) {
-            if (!children[0])
-                return children[0];
-            return signbit(children[0]) ? -0.0 : std::numeric_limits<double>::infinity();
-        }
-        auto ret = getNearestMultiples(children[0], children[1]);
-        return ret.second;
-    }
-    case CalcOperator::Down: {
-        if (children.size() != 2)
-            return std::numeric_limits<double>::quiet_NaN();
-        if (!isinf(children[0]) && isinf(children[1])) {
-            if (!children[0])
-                return children[0];
-            return signbit(children[0]) ? -std::numeric_limits<double>::infinity() : +0.0;
-        }
-        auto ret = getNearestMultiples(children[0], children[1]);
-        return ret.first;
-    }
-    case CalcOperator::Nearest: {
-        if (children.size() != 2)
-            return std::numeric_limits<double>::quiet_NaN();
-        if (!isinf(children[0]) && isinf(children[1]))
-            return signbit(children[0]) ? -0.0 : +0.0;
-        auto ret = getNearestMultiples(children[0], children[1]);
-        auto upperB = ret.second;
-        auto lowerB = ret.first;
-        return std::abs(upperB - children[0]) <= std::abs(children[1]) / 2 ? upperB : lowerB;
-    }
-    case CalcOperator::ToZero: {
-        if (children.size() != 2)
-            return std::numeric_limits<double>::quiet_NaN();
-        if (!isinf(children[0]) && isinf(children[1]))
-            return signbit(children[0]) ? -0.0 : +0.0;
-        auto ret = getNearestMultiples(children[0], children[1]);
-        auto upperB = ret.second;
-        auto lowerB = ret.first;
-        return std::abs(upperB) < std::abs(lowerB) ? upperB : lowerB;
-    }
-    }
-    ASSERT_NOT_REACHED();
-    return 0;
+    return evaluateCalcExpression<double>(calcOperator, children, [](double child) {
```

Comment:
Why can’t the compiler deduce the template argument from the type of `children`?

Reply:
Deduction works here!